airflow/www/views.py

```diff
@@ -1803,54 +1804,104 @@ def _mark_task_instance_state(  # pylint: disable=too-many-arguments
+                downstream=downstream,
+                future=future,
+                past=past,
+                state=state,
+                commit=True,
+                session=session,
+            )
 
-                # Clear downstream tasks that are in failed/upstream_failed state to resume them.
-                # Flush the session so that the tasks marked success are reflected in the db.
-                session.flush()
-                subdag = dag.partial_subset(
-                    task_ids_or_regex={task_id},
-                    include_downstream=True,
-                    include_upstream=False,
-                )
+            # Clear downstream tasks that are in failed/upstream_failed state to resume them.
+            # Flush the session so that the tasks marked success are reflected in the db.
+            session.flush()
+            subdag = dag.partial_subset(
+                task_ids_or_regex={task_id},
+                include_downstream=True,
+                include_upstream=False,
+            )
 
-                end_date = execution_date if not future else None
-                start_date = execution_date if not past else None
-
-                subdag.clear(
-                    start_date=start_date,
-                    end_date=end_date,
-                    include_subdags=True,
-                    include_parentdag=True,
-                    only_failed=True,
-                    session=session,
-                    # Exclude the task itself from being cleared
-                    exclude_task_ids={task_id},
-                )
+            end_date = execution_date if not future else None
+            start_date = execution_date if not past else None
 
-                session.commit()
+            subdag.clear(
+                start_date=start_date,
+                end_date=end_date,
+                include_subdags=True,
+                include_parentdag=True,
+                only_failed=True,
+                session=session,
+                # Exclude the task itself from being cleared
+                exclude_task_ids={task_id},
+            )
 
-            flash(f"Marked {state} on {len(altered)} task instances")
-            return redirect(origin)
+            session.commit()
+
+        flash(f"Marked {state} on {len(altered)} task instances")
+        return redirect(origin)
+
+    @expose('/confirm', methods=['GET'])
+    @auth.has_access(
+        [
+            (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG),
+            (permissions.ACTION_CAN_EDIT, permissions.RESOURCE_TASK_INSTANCE),
+        ]
+    )
+    @action_logging
+    def confirm(self):
+        """Show confirmation page for marking tasks as success or failed."""
+        args = request.args
+        dag_id = args.get('dag_id')
+        task_id = args.get('task_id')
+        execution_date = args.get('execution_date')
+        state = args.get('state')
+
+        upstream = to_boolean(args.get('failed_upstream'))
```

Comment:
I would avoid changing `to_boolean` and do `args.get('failed_upstream', '')` here.

Reply:
IMO it's safe to assume that `None` is `False`. Just saves a lot of code repetition & also safeguard against handling an edge case for the future.